### ISSUE TITLE
Custom URL schemes not supported for URLClassLoader

### ISSUE THREAD

**Issue by chrisr3** [classgraph/classgraph#387]
I have a `URLClassLoader` whose URLs will be a combination of `file://` and a custom `blob://` schemes. A `blob://` is really just an array of bytes containing a single `JAR`, and I create a `ClassGraph` instance from all this as follows:
```kotlin
val blobURL = URL("blob://my/blob")
val appLoader = URLClassLoader(arrayOf(blobURL), null)
val classGraph = ClassGraph()
    .overrideClassLoaders(appLoader)
    .ignoreParentClassLoaders()
    .disableDirScanning()
    .enableRemoteJarScanning()
    .enableExternalClasses()
    .enableAllInfo()
    .verbose(true)
```
However, the `URLClassLoaderHandler` doesn't trigger the "remote jar scanning" mode:
```
2019-11-28T10:01:37.322+0000	ClassGraph	---- Finding unique classloaders in delegation order
2019-11-28T10:01:37.323+0000	ClassGraph	---- Obtaining URLs from classloaders in delegation order
2019-11-28T10:01:37.323+0000	ClassGraph	------ Classloader java.net.URLClassLoader@75000e70 is handled by nonapi.io.github.classgraph.classloaderhandler.URLClassLoaderHandler
2019-11-28T10:01:37.323+0000	ClassGraph	-------- Found classpath element: blob:/my/blob -> /home/chris/workspace/memory-class-graph/blob:/my/blob
2019-11-28T10:01:37.324+0000	ClassGraph	-- Module scanning is disabled, because classloaders or classpath was overridden
2019-11-28T10:01:37.325+0000	ClassGraph	Opening classpath elements (took 0.003030 sec)
2019-11-28T10:01:37.327+0000	ClassGraph	-- Skipping invalid classpath element /home/chris/workspace/memory-class-graph/blob:/my/blob : java.io.FileNotFoundException
2019-11-28T10:01:37.329+0000	ClassGraph	Finding nested classpath elements
2019-11-28T10:01:37.329+0000	ClassGraph	Final classpath element order:
2019-11-28T10:01:37.329+0000	ClassGraph	Scanning classpath elements (took 0.000247 sec)
2019-11-28T10:01:37.329+0000	ClassGraph	Masking classfiles (took 0.000012 sec)
2019-11-28T10:01:37.330+0000	ClassGraph	Scanning classfiles (took 0.000008 sec)
2019-11-28T10:01:37.330+0000	ClassGraph	Linking related classfiles (took 0.000013 sec)
2019-11-28T10:01:37.331+0000	ClassGraph	Total time: 0.006 sec
```
See #389 .

**N.B.** Could there also be a way to restrict _which_ custom URL schemes "remote scanning" is enabled for, please? My `blob://` scheme is really just an array in memory and I wouldn't want accidentally to open the application up to downloading jars via `http://` and `https://` because of it. Perhaps via a new API like?:
```java
.enableRemoteJarScanningFor("blob", ...)
```

**Comment by lukehutch** [comment]
Hey, thanks for the followup report, and for the pull request.

You should be able to use `ClassGraph.enableURLScheme(String scheme)` now. Can you please test if this is fixed for you?

**Comment by chrisr3** [comment]
Thanks, that looks like it worked :smile:.

**Comment by lukehutch** [comment]
Great, thanks for the quick response! I'm working on fixing a performance regression, then will push out another release.

**Comment by lukehutch** [comment]
Fixed and released in 4.8.57.
